src/setup/index.desktop.js

```diff
@@ -1,8 +1,18 @@
 import {AppRegistry} from 'react-native';
+import {ipcRenderer} from 'electron';
+import Onyx from 'react-native-onyx';
 import Config from '../CONFIG';
+import LocalNotification from '../libs/Notification/LocalNotification';
+import ONYXKEYS from '../ONYXKEYS';
+
 
 export default function () {
     AppRegistry.runApplication(Config.APP_NAME, {
         rootTag: document.getElementById('root'),
     });
+
+    ipcRenderer.on('update-downloaded', (_, version) => {
+        LocalNotification.showUpdateAvailableNotification();
```

Comment:
```suggestion
        LocalNotification.showUpdateAvailableNotification(version);
```
Then get rid of the next line that immediately updates Onyx. This way, the modal only appears when the user clicks the notification.

Reply:
I had started with this approach, but the problem I realized when testing locally is that if a user has turned off (or possibly never enabled) notifications for the app, then they will never see this.

Reply:
> I had started with this approach, but the problem I realized when testing locally is that if a user has turned off (or possibly never enabled) notifications for the app, then they will never see this.

Oh, interesting, I hadn't considered that. In that case, I think we would want to just add an `Update` button in the settings page as a backup if there is an update available. You could add that yourself, or we can also just implement that in a follow-up PR, but either way I'd prefer to see the modal only be displayed when the user clicks the notification.

Reply:
I'm going to vote making that a follow-up PR!